src/SFA.DAS.ApprenticeCommitments/Application/DomainEvents/ApprenticeshipChanged.cs

```diff
@@ -0,0 +1,50 @@
+﻿using MediatR;
+using NServiceBus;
+using SFA.DAS.ApprenticeCommitments.Data.Models;
+using SFA.DAS.ApprenticeCommitments.Messages.Events;
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeCommitments.Application.DomainEvents
+{
+    internal class ApprenticeshipChanged : INotification
+    {
+        public Apprenticeship Apprenticeship { get; }
+
+        public ApprenticeshipChanged(Apprenticeship apprenticeship)
+            => Apprenticeship = apprenticeship;
+    }
+
+    internal class ApprenticeshipChangedHandler : INotificationHandler<ApprenticeshipChanged>
+    {
+        private readonly IMessageSession messageSession;
+
+        public ApprenticeshipChangedHandler(IMessageSession messageSession)
+        {
+            this.messageSession = messageSession;
+        }
+
+        public async Task Handle(ApprenticeshipChanged notification, CancellationToken cancellationToken)
+        {
+            var ordered = notification.Apprenticeship.CommitmentStatements
+                .OrderBy(x => x.CommitmentsApprovedOn).ToArray();
+
+            var newest = ordered[^1];
+            var previous = ordered[^2];
+
+            var sinceLastApproval = newest.CommitmentsApprovedOn - previous.CommitmentsApprovedOn;
+            var seenPreviousApproval = notification.Apprenticeship.LastViewed > previous.CommitmentsApprovedOn;
+
+            if (sinceLastApproval > TimeSpan.FromHours(24) || seenPreviousApproval)
```

Comment:
Yes, you're right it should be.

Reply:
The more configuration there is, the more likely we are going to misconfigure something, and I try to walk a line between "the more things that **can** go wrong, the more **will** go wrong" vs "bother, we didn't make this configurable".  

Without prescience it's hard to pull off perfectly.

I spoke to Jon and he is happy for it to remain hard-coded for now.  It would be a small effort in the future to make it configurable, if we ever need to.